### PR TITLE
[SE-0040] Replace '=' with ':' before attribute argument.

### PR DESCRIPTION
**Pull request by dduan**
#### What's in this pull request?

1. Implement [SE-0040](https://github.com/apple/swift-evolution/blob/master/proposals/0040-attributecolons.md). Replace '=' with ':' before a attribute argument. The changes include parser (warning + fixit) and ASTPrinter.
2. Update tests accordingly.
3. Apply new syntax from the proposal to stdlib.

~~Note: at the time this PR is submitted, SE-0040 is still under review. I'm leaving this here _just in case this wild proposal gets accepted_.~~ SE-0040 has been approved.

giphy-1

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: include/swift/AST/Attr.h [hunk 1]
```diff
@@ -1048,10 +1048,10 @@ class SynthesizedProtocolAttr : public DeclAttribute {
 ///
 /// \code
 /// struct X {
-///   @warn_unused_result(message="this string affects your health")
+///   @warn_unused_result(message:"this string affects your health")
```
Comment (gribozavr) [inline include/swift/AST/Attr.h]: I think we want a space after the colon.

**Comment by gribozavr** [comment]
@swift-ci Please test

File: include/swift/AST/Attr.h (hunk 1 above)
Comment (dduan) [inline include/swift/AST/Attr.h]: That is the case in ASTPrinter and stdlib. I'll update this doc.

Comment (dduan) [inline include/swift/AST/Attr.h]: But I think that would hide CI results. So after that.

**Comment by lattner** [comment]
Heh, thanks.  It certainly seems like a highly controversial proposal! :-)

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by dduan** [comment]
Help! These patches have big impact areas and I have been resolving lots of conflicts. It'd be great if they get a quick review and be merged soon. 🙇

**Comment by lattner** [comment]
Awesome, merged and tests pass for me, thank you!

**Comment by dduan** [comment]
Great. Thanks Chris!

**Pull request merged**